rflx/pyrflx/typevalue.py

```diff
@@ -224,34 +237,221 @@ class OpaqueValue(TypeValue):
     def __init__(self, vtype: Opaque) -> None:
         super().__init__(vtype)
 
-    def assign(self, value: bytes, check: bool = True) -> None:
+    def assign(self, value: bytes, length: int = 0, check: bool = True) -> None:
         self._value = value
 
-    def assign_bitvalue(self, value: str, check: bool = True) -> None:
+    def assign_bitvalue(self, value: Bitstring, length: int = 0) -> None:
+        self._value = bytes(value)
 
-        while len(value) % 8 != 0:
-            value = "0" + value
+    @property
+    def length(self) -> int:
+        self._raise_initialized()
+        return len(self._value) * 8
 
-        bytestring = b"".join(
-            [int(value[i : i + 8], 2).to_bytes(1, "big") for i in range(0, len(value), 8)]
-        )
+    @property
+    def value(self) -> bytes:
+        self._raise_initialized()
+        return self._value
+
+    @property
+    def to_bitstring(self) -> Bitstring:
+        self._raise_initialized()
+        return Bitstring(format(int.from_bytes(self._value, "big"), f"0{self.length}b"))
+
+    @property
+    def accepted_type(self) -> type:
+        return bytes
+
+    @property
+    def literals(self) -> Mapping[Name, Expr]:
+        return {}
+
+
+class ArrayValue(TypeValue):
+
+    _value: List[TypeValue]
+
+    def __init__(self, vtype: Array) -> None:
+        super().__init__(vtype)
+        self._element_type = vtype.element_type
+        self._is_message_array = isinstance(self._element_type, model.Message)
+
+    def assign(self, value: List[TypeValue], length: int = 0, check: bool = True) -> None:
+
+        # check if all elements are of the same class and messages are valid
+        for v in value:
+            if (
+                self._is_message_array
+                and isinstance(v, MessageValue)
+                and isinstance(self._element_type, model.Message)
+            ):
+                if not v.equal_model(self._element_type):
+                    raise ValueError("members of an array must not be of different classes")
+                if not v.valid_message:
+                    raise ValueError("cannot assign array of messages: messages must be valid")
+            else:
+                if not self._is_message_array and not v.equal_type(self._element_type):
+                    raise ValueError("members of an array must not be of different classes")
+
+        self._value = value
+
+    def assign_bitvalue(self, value: Bitstring, length: int = 0) -> None:
```

Comment:
Yes it is. Do you want me to rename it? I thought it would be good for consistency if all TypeValues had an assign_bitvalue method. IMHO a parse method for an IntegerValue would not be very intuitive either.

Reply:
All TypeValue should have the same method. I just think `assign_bitvalue` is not a very intuitive name. It is clear to me that one can assign an `int` to an `IntegerValue`, but assigning a `Bitstring` is kinda strange to me. IMHO `parse` would fit better here.
@jklmnn: What is your opinion?

Reply:
I think `parse` would be a good alternative.

Reply:
Fixed by a1d50d6